src/main/java/gwt/jsonix/marshallers/xjc/plugin/builders/MainJsBuilder.java

```diff
@@ -134,16 +139,20 @@ protected static void populateJCodeModel(JCodeModel toPopulate, JClass container
         addMarshall(toPopulate, mainJsClass, containerRef, callbackMap.get(MARSHALL_CALLBACK));
     }
 
-    protected static JDefinedClass getMainJsClass(final JCodeModel toPopulate, String basePackage) throws JClassAlreadyExistsException {
-        String fullMainJsName = basePackage + "." + MAIN_JS;
+    protected static JDefinedClass getMainJsClass(final JCodeModel toPopulate, String basePackage, String mainJsName) throws JClassAlreadyExistsException {
```

Comment:
as we touched the header, could we use format as we do for `generateJSInteropMainJs` please?
```java
protected static JDefinedClass getMainJsClass(final JCodeModel toPopulate,
                                                  final String basePackage,
                                                  final String mainJsName) throws JClassAlreadyExistsException {
```

Reply:
@jomarko 
done